packages/react/src/components/OverflowMenu/OverflowMenu.tsx

```diff
@@ -92,8 +116,123 @@ export const getMenuOffset = (menuBody, direction, trigger, flip) => {
   }
 };
 
-class OverflowMenu extends Component {
-  state = {};
+type MaybeArray<T> = T | T[];
+type MaybeGetter<T> = T | (() => T);
+type Offset = { top: number; left: number };
+type IconProps = {
+  className?: string;
+  'aria-label'?: string;
+};
+
+type ChildElement = ReactElement<
+  OverflowMenuItemProps,
+  typeof OverflowMenuItem
+>;
+
+export interface OverflowMenuProps extends HTMLAttributes<HTMLElement> {
+  /**
+   * The child nodes.
+   */
+  children: MaybeArray<ChildElement>;
```

Comment:
We can't restrict `children` to only be OverflowMenuItem(s).

```suggestion
  children?: React.ReactNode;
```